### PR TITLE
treewide: modernize python packages

### PR DESCRIPTION
**Pull request by pbsds**
## Description of changes

Continuation of https://github.com/NixOS/nixpkgs/pull/334019, I grepped for --replace and:

* use `pyproject = true;`
* use `build-system` and `dependencies`
* use `--replace-fail`, and fix failures by or either changing, deleting or using other means like `pytest-cov-stub` and `pythonRelaxDeps`
* remove `repo = pname;`

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/pyverilog/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/pyverilog/default.nix:28]:
```suggestion
      --replace-fail "iverilog = 'iverilog'" "iverilog = '${lib.getExe' iverilog "iverilog"}'"
```

File: pkgs/development/python-modules/snakemake-storage-plugin-xrootd/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/snakemake-storage-plugin-xrootd/default.nix:27]: use `pythonRemoveDeps`

File: pkgs/development/python-modules/spyse-python/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/spyse-python/default.nix:43]: Use `pythonRelaxDeps` and `pythonRemoveDeps`

File: pkgs/development/python-modules/testing-postgresql/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/testing-postgresql/default.nix:51]: use `pythonRelaxDeps`

File: pkgs/development/python-modules/yowsup/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/yowsup/default.nix:36]:
```suggestion
  pythonRelaxDeps = true;

  pythonRemoveDeps = [ "argparse" ];
```

File: pkgs/development/python-modules/zeversolarlocal/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/zeversolarlocal/default.nix:58]: This can be removed.

File: pkgs/development/python-modules/pymoo/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/pymoo/default.nix:52]: use `pythonRelaxDeps` and `pythonRemoveDeps`

File: pkgs/development/python-modules/pyosohotwaterapi/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/pyosohotwaterapi/default.nix:31]: use `pythonRemoveDeps`

File: pkgs/development/python-modules/pytest-ansible/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/pytest-ansible/default.nix:33]:
```suggestion
      --replace-fail '/usr/bin/env' '${lib.getExe' coreutils "env"}'
```

File: pkgs/development/python-modules/pytest-raisesregexp/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/pytest-raisesregexp/default.nix:17]:
```suggestion
    hash = "...";
```

File: pkgs/development/python-modules/pyisbn/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/pyisbn/default.nix:21]:
```suggestion
    hash = "...";
```

File: pkgs/development/python-modules/pymetar/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/pymetar/default.nix:19]:
```suggestion
    hash = "...";
```

File: pkgs/development/python-modules/sanic-auth/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/sanic-auth/default.nix:19]:
```suggestion
    hash = "...";
```

File: pkgs/development/python-modules/snakebite/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/snakebite/default.nix:16]:
```suggestion
    hash = "...";
```

File: pkgs/development/python-modules/videocr/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/videocr/default.nix:19]:
```suggestion
    hash = "...";
```

File: pkgs/development/python-modules/zeversolarlocal/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/zeversolarlocal/default.nix:25]:
```suggestion
    hash = "...";
```

**Review by philiptaron — APPROVED** [review]
I'm neutral on dotlambda's suggestions; I have every confidence in another tree wide to sort them out.

**Comment by pbsds** [comment]
I left the lower version bounds in spyse-python untouched, i'll let package maintainers to decide to remove those.
xrootd was relaxed rather than removed.

I'll have to revive https://github.com/NixOS/nixpkgs/pull/318290 at some point when i'm done traveling across the globe, i think you'd be interested @dotlambda

**Comment by pbsds** [comment]
i had to revert the `pyproject` migration for python312Packages.py-libzfs, since it fails the build with a missing header file include. Likely the pypaBuildHook doesn't runs some configuration or something

**Review by fpletz — APPROVED** [review]

**Pull request merged**
